core/src/main/java/nl/nn/adapterframework/receivers/Receiver.java

```diff
@@ -523,7 +527,11 @@ public void configure() throws ConfigurationException {
 			super.configure();
 			if (StringUtils.isEmpty(getName())) {
 				if (getListener()!=null) {
-					setName(ClassUtils.nameOf(getListener()));
+					if(StringUtils.isNotEmpty(getListener().getName())) {
+						setName(getListener().getName() + " " + ClassUtils.nameOf(getListener()));
+					} else {
+						setName(ClassUtils.nameOf(getListener()));
+					}
 				} else {
 					setName(ClassUtils.nameOf(this));
```

Comment:
I think it would be wise to check for other Receivers (in the same Adapter) that have the same name, and if they exist, disambiguate by adding a sequence number